### PR TITLE
Struct and OpenStruct will be able to chain the Hash#dig, Array#dig

### PR DESCRIPTION
**Pull request by kachick**

### CODE REVIEW

**Comment by kachick** [comment]
ticket: https://redmine.ruby-lang.org/issues/11688

File: lib/ostruct.rb [hunk 1]
```diff
@@ -215,6 +215,20 @@ def []=(name, value)
...
+  # Retrieves the value object corresponding to the each +name+
+  # objects repeatedly.
+  #
+  #   address = OpenStruct.new('city' => "Anytown NC", 'zip' => 12345)
+  #   person = OpenStruct.new('name' => 'John Smith', 'address' => address)
+  #   person.dig(:address, 'zip') # => 12345
+  #   person.dig(:business_address, 'zip') # => nil
+  #
+  def dig(*names)
+    names.map!(&:to_sym)
```
Comment (0x0dea) [inline lib/ostruct.rb]: Is it not unwise to mutate this argument?

**Comment by hsbt** [comment]
merged at 482530680c18ea2e44c3300c6f323fabc3bd55f7

**Comment by kachick** [comment]
Thanks to merge and improvements!

**Comment by marcandre** [comment]
FYI, I changed the behavior for arguments that are not symbol-like, as per https://bugs.ruby-lang.org/issues/11762

**Comment by nobu** [comment]
Thank you, I've forgot it.

**Comment by kachick** [comment]
@marcandre :memo: :bow:

**Pull request closed without merging**